src/merge_bot/merge_bot.py

```diff
@@ -57,54 +57,63 @@ def configure_commit_info(repo, bot_name, bot_email):
         config.set_value("user", "name", bot_name)
 
 
+def git_is_ancestor(gitwd, a, b):
+    try:
+        gitwd.git.merge_base(a, b, is_ancestor=True)
+        return True
+    except git.exc.GitCommandError:
+        # merge_base --is-ancestor indicates true/false by raising an
+        # exception or not
+        return False
+
+
 def git_merge(gitwd, dest, source, merge):
     orig_commit = gitwd.active_branch.commit
 
-    if merge.branch in gitwd.remotes.merge.refs:
-        # Check if we have already pushed a merge to the merge branch which
-        # contains the current head of the source branch
+    def do_merge():
```

Comment:
Interesting. Why do you wrap this code in a nested function and immediately call it after? Is it to better convey the meaning of this code? Not advocating against the change itself, but it doesn't belong to this commit as it makes it harder to review the functional change.

Reply:
It's syntactic sugar when you want to shortcut out of a conditional block. It's much simpler to return than to nest a dozen ifs.

Reply:
This it what nested functions are generally used for, but this isn't what you're doing here, are you?
Since you're calling `do_merge()` immediately after defining the inner function, and only here, you're effectively only adding one indentation level for this code. Hence my question about the purpose of this change.